src/test/java/com/kaltura/client/test/tests/servicesTests/transactionHistoryTests/ListTests.java

```diff
@@ -0,0 +1,119 @@
+        PurchaseUtils.purchasePpv(userKs, Optional.of(getSharedMediaAsset().getId().intValue()), Optional.of(assetFileId), null);   //Purchase asset
+        String ppvModuleId = PurchaseUtils.purchasePpvDetailsMap.get("ppvModuleId");
+        String pricePpv = PurchaseUtils.purchasePpvDetailsMap.get("price");
+        String currencyPpv = PurchaseUtils.purchasePpvDetailsMap.get("currency");
+
+        entitlementPpvFilter.setProductTypeEqual(TransactionType.PPV);  //Show only PPV purchases
+        ListEntitlementBuilder listEntitlementBuilder = EntitlementService.list(entitlementPpvFilter).setKs(userKs);
+        listEntitlementServiceResponse = executor.executeSync(listEntitlementBuilder);  //entitlement/action/list for household
+        List<PpvEntitlement> ppvEntitlementList = new ArrayList<>();    //Conversion from Entitlement to PpvEntitlement
+        for(Entitlement entitlement:listEntitlementServiceResponse.results.getObjects()){
+            if(entitlement.getClass() == PpvEntitlement.class) {
+                ppvEntitlementList.add((PpvEntitlement) entitlement);
+            }
+        }
+        assertThat(listEntitlementServiceResponse.results.getTotalCount()).isEqualTo(1);    //Verify that count of entitlements is 1
+        assertThat(ppvEntitlementList.get(0).getProductId()).isEqualTo(ppvModuleId);    //Verify that PPV is correct
+        assertThat(ppvEntitlementList.get(0).getMediaFileId()).isEqualTo(assetFileId);  //Verify that asset file is correct
+
+        transactionHistoryFilter.setEntityReferenceEqual(EntityReferenceBy.USER);   //All transactions per user
+        listTransactionHistoryBuilder = TransactionHistoryService.list(transactionHistoryFilter).setKs(userKs); //Use non-master user ks
+        listBillingTransactionResponse = executor.executeSync(listTransactionHistoryBuilder);   //transactionHistory/action/list for non-master user
+        assertThat(listBillingTransactionResponse.error).isNull();  //No errors appeared at response
+        assertThat(listBillingTransactionResponse.results.getTotalCount()).isEqualTo(1);    //Verify that one transaction was performed
+        assertThat(listBillingTransactionResponse.results.getObjects().get(0).getPurchasedItemCode()).isEqualTo(getSharedMediaAsset().getId().toString());  //Verify asset id is correct
+        assertThat(listBillingTransactionResponse.results.getObjects().get(0).getItemType().getValue()).isEqualTo(TransactionType.PPV.getValue()); //Verify transaction is for PPV
+        assertThat(listBillingTransactionResponse.results.getObjects().get(0).getPrice().getAmount().toString()).isEqualTo(pricePpv);   //Verify price amount
+        assertThat(listBillingTransactionResponse.results.getObjects().get(0).getPrice().getCurrency().toString()).isEqualTo(currencyPpv);  //Verify price currency
+        assertThat(listBillingTransactionResponse.results.getObjects().get(0).getBillingPriceType().getValue()).isEqualTo("FullPeriod");    //Verify entitlement period
```

Comment:
Please use already created ENUMs:
instead of assertThat(listBillingTransactionResponse.results.getObjects().get(0).getBillingPriceType().getValue()).isEqualTo("FullPeriod"); 

use 

assertThat(listBillingTransactionResponse.results.getObjects().get(0).getBillingPriceType()).isEqualTo(BillingPriceType.FULLPERIOD);